dotnetv3/cross-service/PhotoAssetManager/PamApi/DownloadFunction.cs

```diff
@@ -0,0 +1,90 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier:  Apache-2.0
+
+using Amazon;
+using Amazon.DynamoDBv2;
+using Amazon.DynamoDBv2.DataModel;
+using Amazon.Lambda.Core;
+using Amazon.S3;
+using Amazon.SimpleNotificationService;
+using Amazon.Util;
+using PamServices;
+using System.Text.Json;
+
+namespace PamApi;
+
+/// <summary>
+/// The function code for downloading and zipping
+/// images from a list of labels.
+/// </summary>
+public class DownloadFunction
+{
+    IAmazonS3 S3Client { get; set; }
+    private readonly StorageService _storageService;
+    private readonly NotificationService _notificationService;
+    private readonly LabelService _labelService;
+
+    /// <summary>
+    /// Default constructor. This constructor is used by Lambda to construct the instance. When invoked in a Lambda environment
+    /// the AWS credentials will come from the IAM role associated with the function and the AWS region will be set to the
+    /// region the Lambda function is executed in.
+    /// </summary>
+    public DownloadFunction()
+    {
+        S3Client = new AmazonS3Client();
+        _notificationService = new NotificationService(new AmazonSimpleNotificationServiceClient());
+        _storageService = new StorageService(S3Client);
+        _labelService = new LabelService(new DynamoDBContext(new AmazonDynamoDBClient()));
+        var labelsTableName = Environment.GetEnvironmentVariable("LABELS_TABLE_NAME");
+
+        if (labelsTableName != null)
+        {
+            AWSConfigsDynamoDB.Context.AddMapping(new TypeMapping(typeof(Label), labelsTableName));
+        }
```

Comment:
Any concerns with silent failures here if `LABELS_TABLE_NAME` is missing?

Reply:
It won't fail silently, it will fail with an error that that table wasn't found. I know because I did that a few times by mistake!